wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos/kotlin/alltypes/AllTypes.kt

```diff
@@ -2491,6 +2671,126 @@ public class AllTypes(
+                  .toInt()
+                array_int32 = IntArrayList(initialCapacity)
+              }
+              array_int32!!.add(com.squareup.wire.ProtoAdapter.INT32.decodePrimitive(reader))
+            }
+            602 -> {
+              if (array_uint32 == null) {
+                val minimumByteSize = 1
+                val initialCapacity = (reader.nextFieldMinLengthInBytes() / minimumByteSize)
+                  .coerceAtMost(Int.MAX_VALUE.toLong())
+                  .toInt()
+                array_uint32 = IntArrayList(initialCapacity)
+              }
+              array_uint32!!.add(com.squareup.wire.ProtoAdapter.UINT32.decodePrimitive(reader))
+            }
+            603 -> {
+              if (array_sint32 == null) {
+                val minimumByteSize = 1
+                val initialCapacity = (reader.nextFieldMinLengthInBytes() / minimumByteSize)
+                  .coerceAtMost(Int.MAX_VALUE.toLong())
+                  .toInt()
+                array_sint32 = IntArrayList(initialCapacity)
+              }
+              array_sint32!!.add(com.squareup.wire.ProtoAdapter.SINT32.decodePrimitive(reader))
+            }
+            604 -> {
+              if (array_fixed32 == null) {
+                val minimumByteSize = 4
+                val initialCapacity = (reader.nextFieldMinLengthInBytes() / minimumByteSize)
+                  .coerceAtMost(Int.MAX_VALUE.toLong())
+                  .toInt()
+                array_fixed32 = IntArrayList(initialCapacity)
+              }
+              array_fixed32!!.add(com.squareup.wire.ProtoAdapter.FIXED32.decodePrimitive(reader))
+            }
+            605 -> {
+              if (array_sfixed32 == null) {
+                val minimumByteSize = 4
+                val initialCapacity = (reader.nextFieldMinLengthInBytes() / minimumByteSize)
+                  .coerceAtMost(Int.MAX_VALUE.toLong())
+                  .toInt()
+                array_sfixed32 = IntArrayList(initialCapacity)
+              }
+              array_sfixed32!!.add(com.squareup.wire.ProtoAdapter.SFIXED32.decodePrimitive(reader))
+            }
+            606 -> {
+              if (array_int64 == null) {
+                val minimumByteSize = 1
+                val initialCapacity = (reader.nextFieldMinLengthInBytes() / minimumByteSize)
+                  .coerceAtMost(Int.MAX_VALUE.toLong())
+                  .toInt()
+                array_int64 = LongArrayList(initialCapacity)
+              }
+              array_int64!!.add(com.squareup.wire.ProtoAdapter.INT64.decodePrimitive(reader))
+            }
+            607 -> {
+              if (array_uint64 == null) {
+                val minimumByteSize = 1
+                val initialCapacity = (reader.nextFieldMinLengthInBytes() / minimumByteSize)
+                  .coerceAtMost(Int.MAX_VALUE.toLong())
+                  .toInt()
+                array_uint64 = LongArrayList(initialCapacity)
```

Comment:
Should `LongArrayList` have a constructor that takes a length in bytes?

Reply:
(that’d reduce the total amount of generated code)

Maybe even a factory function:

```
LongArrayList.forDecoding(minLengthInBytes: Long)
```